plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1361,6 +1361,12 @@ public void setColumnType(ConnectorSession session, ConnectorTableHandle tableHa
         failIfAvroSchemaIsSet(table);
         HiveColumnHandle column = (HiveColumnHandle) columnHandle;
 
+        HiveStorageFormat storageFormat = extractHiveStorageFormat(metastore.getTable(table.getSchemaName(), table.getTableName())
+                .orElseThrow(() -> new TableNotFoundException(table.getSchemaTableName())));
+        if (storageFormat != HiveStorageFormat.ORC && storageFormat != HiveStorageFormat.PARQUET) {
+            throw new TrinoException(NOT_SUPPORTED, "Unsupported storage format for changing column type: " + storageFormat);
```

Comment:
is the check effective?
can a partitioned table have partitions with different file format?

Reply:
btw it's looks like we should be able to lift the limitation for eg TEXTFILE easily.
please add a code comment explaining the logic, why ORC/PARQUET only and what is our stance about partitioned tables.